wordpress-mobile/WordPress-iOS
# Add firebase for ad source attribution

Fixes #8426

This ads the Firebase library to the project, mostly to be able to track the source of downloads that are due to AdWords ads. In the future Firebase and Crashlytics will merge into just Firebase (apparently), and at that point we'll be able to remove Crashlytics.

*Note*: 
Client ID and API values are not stored in the project, and are added at runtime. They are `GOOGLE_FIREBASE_APP_ID`, `GOOGLE_FIREBASE_API_KEY`, and `GOOGLE_FIREBASE_GCM_ID`. As such, Firebase dumps this error to console at every launch:

```
[Firebase/Core][I-COR000012] Could not locate configuration file: 'GoogleService-Info.plist'.
2018-01-23 23:18:22.159509-0600 WordPress[10223:542972] 4.8.1 - [Firebase/Analytics][I-ACS020006] Google App ID from GoogleService-Info.plist is empty. Please, define GOOGLE_APP_ID in GoogleService-Info.plist for Analytics to work reliably. See https://goo.gl/txkZbE
2018-01-23 23:18:22.162600-0600 WordPress[10223:542996] 4.8.1 - [Firebase/Analytics][I-ACS025020] Analytics requires Google App ID from GoogleService-Info.plist. Your data may be lost.
```

This appears to be perfectly acceptable, as `GOOGLE_APP_ID` is set just moments later.

To test:
- add the three values mentioned above
- launch the app
- check if the session appears on the Firebase console

Would love to hear of a way to test ad attribution before release.

I would also like build/config feedback — @elibud @diegoreymendez
